composer/utils/checkpoint.py

```diff
@@ -449,7 +466,13 @@ def save_checkpoint(
     paths = dist.all_gather_object(checkpoint_filepath)
     paths = list(pathlib.Path(path) for path in paths if path is not None)
 
-    return paths
+    # TODO: refactor this function
```

Comment:
what does this mean? refactor what part?

Reply:
Will refactor this entire function in a follow-on PR :)